src/composables/useRaycaster/index.ts

```diff
@@ -20,10 +20,10 @@ interface PointerClickEventPayload {
 
 export const useRaycaster = (
   objects: Ref<THREE.Object3D[]>,
-  { renderer, camera, raycaster }: Pick<TresContext, 'renderer' | 'camera' | 'raycaster'>,
+  ctx: TresContext,
```

Comment:
Hey Tino, thanks for the feedback. While I understand the reasoning behind being verbose about the deps, don't you consider it overcomplicated when you read it through? 

> because some parts of the context might not exist yet.

deconstructing needs the same check as `ctx.camera === undefined` in case it doesn't exist yet.

In this specific instance, I opted for passing the entire TresContext (ctx) for a few reasons:

1. **Simplicity**: The function in question utilizes multiple properties of `TresContext`. Passing the entire context rather than destructuring simplifies the function signature.

2. **Future-proofing**: If we need to use more properties of `TresContext` in the future, we can avoid modifying the function signature each time, reducing the potential for errors and merge conflicts.

3. **Performance Consideration**: Yep, it's not a significant impact in most cases, avoiding destructuring means we don't create additional variables unnecessarily, which is a marginal optimization.

We need also to consider what would be easier to maintain, especially if we want to delegate some level of maintenance to the community.